flytekit/tools/translator.py

```diff
@@ -160,8 +160,10 @@ def fn(settings: SerializationSettings) -> List[str]:
 
 
 def get_serializable_task(
+    entity_mapping: OrderedDict,
```

Comment:
Where do you pass this argument?

Reply:
Line 701 in `get_serializable`. There are similar method for workflows, launchplans, etc that all take in this argument so that they can recursively register entities. Now we enable tasks to do the same if they are dynamic and they have a `output_entity_hint`

Reply:
Oh thanks, Github truncated and I didn't realise

Reply:
Ok, so this code will mutate the `entity_mapping` ordered dict by adding a task with an entity attached to it thanks to the type hint? Then this ordered dict gets iterated through and everything in it gets registered in a topological order?

Reply:
Technically the entity is attached based on its actual type with `isinstance` with recursive calls to `get_serializable`. It doesn't actually use a type hint. Other than that its as you describe.